src/utils.js

```diff
@@ -23,6 +23,11 @@ const ABORT_ERR_NAME = 'AbortError';
 /** DOMException.ABORT_ERR = 20 */
 const ABORT_ERR_CODE = 20;
 
+/** Normal protocol shape is "xyz:" */
+const PROTOCOL_RE = /.*\:$/;
+
+const SCHEME_REDIRECT_BLACKLIST_RE = /javascript|script|data/i;
```

Comment:
Why can't this be an allow-list instead of a disallow-list?

Reply:
Not super comfortable with this either. But I wanted to enable custom protocols (in- and out-of-native apps) by default. There's a separate `accept` API that requires more conformance before the actual data is sent over. Thus, in this form, this API is mainly for cancelations and errors.